test/uk/gov/hmrc/pegaproofofconceptfrontend/controllers/CallbackControllerSpec.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pegaproofofconceptfrontend.controllers
+
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.http.Status
+import play.api.inject.bind
+import play.api.inject.guice.GuiceableModule
+import play.api.libs.json.JsNull
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{contentAsString, defaultAwaitTimeout, status}
+import uk.gov.hmrc.auth.core.AuthConnector
+import uk.gov.hmrc.auth.core.authorise.Predicate
+import uk.gov.hmrc.auth.core.retrieve.Retrieval
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.test.ExternalWireMockSupport
+import uk.gov.hmrc.pegaproofofconceptfrontend.testsupport.FakeApplicationProvider
+import uk.gov.hmrc.pegaproofofconceptfrontend.utils.Generators
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CallbackControllerSpec extends AnyWordSpec with Matchers with GuiceOneAppPerSuite with ExternalWireMockSupport with FakeApplicationProvider
+  with Generators with ScalaCheckDrivenPropertyChecks {
+
+  private val fakeAuthConnector = new AuthConnector {
+    override def authorise[A](predicate: Predicate, retrieval: Retrieval[A])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[A] = {
+      Future.successful(JsNull.as[A](retrieval.reads))
+    }
+  }
+
+  override val overrideModules: Seq[GuiceableModule] = Seq(bind[AuthConnector].toInstance(fakeAuthConnector))
+
+  private val controller = app.injector.instanceOf[CallbackController]
+
+  private val fakeRequest = FakeRequest()
+
+  "returns" should {
```

Comment:
we could have a small test for the callback endpoint as well - it would just call the endpoint and check we get a redirect happening to the right place

Reply:
looking into it, but I'm unsure how to test that atm

Reply:
it would roughly look like this I think:
```scala
import play.api.test.Helpers._ // I think this is the import...

val result = controller.callback("dummyQueryParameter")(fakeRequest)
status(result) shouldBe SEE_OTHER
redirectLocation(result) shouldBe Some("expected-redirect-url")
```

Reply:
I was wondering if it should be a 303; good to know that is the case. ty